internal/gatewayapi/filters.go

```diff
@@ -764,6 +764,30 @@ func (t *Translator) processExtensionRefHTTPFilter(extFilter *v1beta1.LocalObjec
 		}
 	}
 
+	// This list of resources will be empty unless an extension is loaded (and introduces resources)
+	for _, res := range resources.ExtensionRefFilters {
```

Comment:
imo validation in the gateway API layer is useful to check whether the extFilter within a `HTTPRoute` points to an existing extension resource. This code loops over `resources.ExtensionRefFilters` rather than checking whether the filter is an extension type and whether the typed instance exists or not

Reply:
Let me know if my following understanding sounds correct to you or if I missed your point here and need to fix something.

So there are two different groups for the "validity" of extensionRef filters. The first is whether the group/kind is supported or not:
1. Group/Kind is natively supported (Authentication/RateLimit)
2. Group/Kind is supported by extensions
3. Group/Kind is not supported

The second is whether the resource exists or not (we don't get to this check unless the group/kind was already confirmed to be supported):
1. Reference is found
2. Reference is not found

>This code loops over resources.ExtensionRefFilters rather than checking whether the filter is an extension type and whether the typed instance exists or not

This logic in `processExtensionRefHTTPFilter()` is taking a `LocalObjectReference` and the list of loaded resources that EG is watching to see if the referenced object exists or not, so the section I added should be doing a similar job to the bits of code above it that check the Auth / Ratelimit resources.

`ValidateHTTPRouteFilter()` is what performs the check you mentioned on whether the group/kind is supported or not and it gets executed before we run `processExtensionRefHTTPFilter()`. If the group/kind is not supported then we break out early and do not execute `processExtensionRefHTTPFilter()`.

`resources.ExtensionRefFilters` only contains resources that we are watching for an extension, so it should never contain unsupported resources, now we just need to confirm if the resource that was referenced exists in there or not if it isn't an Auth/Ratelimit filter that is natively supported.

For Auth/Ratelimit we can first just check if the `extFilter`'s type matches the Auth/Ratelimit types since they are static and we know what they are so we can just check `resources.AuthenticationFilters` and `resources.RateLimitFilters` respectively. For extension resources, we don't know what all the types stored in `resources.ExtensionRefFilters` are, but we do know for certain that `resources.ExtensionRefFilters` only contains resources owned by the extension. So that's why I had it iterate over the resources first and just make sure that the name/namespace/group/kind match the `LocalObjectReference`.

I added another test case file just in case to make sure that even if `resources.extensionRefFilters` contains a referenced resource that is not supported by extensions (it never should since it only contains watched extension resources), we still set `UnsupportedValue`: `Invalid filter ExtensionRef: unknown kind foo.example.io/Unsupported` rather than `ResolvedRefs`: `Reference default/example not found for filter type: Foo`.